File: Selenium/Project/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
import time

# Set up the Firefox Driver with WebDriverManager
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the OrangeHRM page
    driver.get("http://alchemy.hguy.co/orangehrm")

    # Login with credentials
    username = driver.find_element(By.ID, "txtUsername")
    username.send_keys("orange")

    password = driver.find_element(By.ID, "txtPassword")
    password.send_keys("orangepassword123")

    login_btn = driver.find_element(By.ID, "btnLogin")
    login_btn.click()

    # Find the "My Info" menu item and click it
    my_info_menu = driver.find_element(By.ID, "menu_pim_viewMyDetails")
    my_info_menu.click()

    # Find the Qualification option on the left side menu and click it
    qualifications_option = driver.find_element(By.XPATH, "//a[text()='Qualifications' and not(@class='arrow')]")
    qualifications_option.click()

    # Add Work Experience and click Save
    add_work_experience_btn = driver.find_element(By.ID, "addWorkExperience")
    add_work_experience_btn.click()

    # Fill in Work Experience details
    company_name = driver.find_element(By.ID, "experience_employer")
    company_name.send_keys("ABC Company")

    job_title = driver.find_element(By.ID, "experience_jobtitle")
    job_title.send_keys("Software Engineer")

    from_date = driver.find_element(By.ID, "experience_from_date")
    from_date.send_keys("2024-01-01")

    to_date = driver.find_element(By.ID, "experience_to_date")
    to_date.send_keys("2024-12-31")

    # Save the work experience
    save_btn = driver.find_element(By.ID, "btnWorkExpSave")
    save_btn.click()

    # Wait for a while to see the result
    time.sleep(5)

    driver.close()

File: Selenium/Project/Activity7.java
package OrangeHRM;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.Reporter;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import java.time.Duration;
import java.util.List;

public class Activity7 {

        WebDriver driver;
        WebDriverWait wait;

        @BeforeTest(alwaysRun = true)
        public void setup() {
            // Set up Firefox driver
            WebDriverManager.firefoxdriver().setup();
            //Create a new instance of the Firefox driver
            driver = new FirefoxDriver();
            wait = new WebDriverWait(driver, Duration.ofSeconds(25));
            // Open the page
            driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");

            Reporter.log("Opened Browser |");
        }
        @Test(priority=0)
        public void loginTestCase() {
            //Find username and pasword fields
            WebElement usernameField = driver.findElement(By.id("txtUsername"));
            WebElement passwordField = driver.findElement(By.id("txtPassword"));

            //Enter values
            usernameField.sendKeys("orange");
            passwordField.sendKeys("orangepassword123");

            //Click Log in
            driver.findElement(By.id("btnLogin")).click();
            String pagetitle= driver.getTitle();
            System.out.println("pagetitle is: "+pagetitle);
            Assert.assertEquals(pagetitle, "OrangeHRM");
        }
        @Test(priority = 1)
        public void addqualifications(){
            driver.findElement(By.id("menu_pim_viewMyDetails")).click();
            wait = new WebDriverWait(driver, Duration.ofSeconds(15));
            driver.findElement(By.xpath("//a[text()='Qualifications' and not(@class='arrow')]/parent::li")).click();
            driver.findElement(By.id("addWorkExperience")).click();
            driver.findElement(By.id("experience_employer")).sendKeys("acf");
            driver.findElement(By.id("experience_jobtitle")).sendKeys("test");
            driver.findElement(By.id("btnWorkExpSave")).click();
            Reporter.log("Success message displayed");
        }
        @AfterTest(alwaysRun = true)
        public void closeBrowser() {
            //Close the browser
            driver.close();
        }
    }
